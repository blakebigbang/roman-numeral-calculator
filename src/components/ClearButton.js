import React from "react";

import Button from "./Button";
import ExperimentalButton from "./ExperimentalButton";

import UI from "../utils/UI";

export const handleClear = ({
  vibrate,
  setError,
  first,
  second,
  setPreviousOperation,
  operationMode,
  setFirst,
  setOperationMode,
  setSecond
}) => {
  vibrate();
  setError(false);

  if (!first && !second) {
    return setPreviousOperation();
  } else if (first && !operationMode && !second) {
    return setFirst("");
  } else if (first && operationMode && !second) {
    return setOperationMode();
  } else if (first && !operationMode && second) {
    // impossible
  } else if (first && operationMode && second) {
    return setSecond("");
  }
};

function ClearButton({
  theme,
  vibrate,
  setError,
  first,
  second,
  setPreviousOperation,
  operationMode,
  setFirst,
  setOperationMode,
  setSecond
}) {
  if (theme === UI.THEMES_KEYS.CLASSIC) {
    return (
      <Button
        buttonType={UI.BUTTON_TYPES.EDIT}
        tabIndex={0}
        onClick={() =>
          handleClear({
            vibrate,
            setError,
            first,
            second,
            setPreviousOperation,
            operationMode,
            setFirst,
            setOperationMode,
            setSecond
          })
        }
      >
        AC
      </Button>
    );
  } else if (theme === UI.THEMES_KEYS.EXPERIMENTAL) {
    return (
      <ExperimentalButton
        buttonType={UI.BUTTON_TYPES.EDIT}
        tabIndex={0}
        onClick={() =>
          handleClear({
            vibrate,
            setError,
            first,
            second,
            setPreviousOperation,
            operationMode,
            setFirst,
            setOperationMode,
            setSecond
          })
        }
      >
        AC
      </ExperimentalButton>
    );
  }
  return null;
}

export default ClearButton;
