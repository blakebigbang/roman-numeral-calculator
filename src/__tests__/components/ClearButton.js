import React from "react";
import renderer from "react-test-renderer";

import ClearButton from "../../components/ClearButton";

describe("ClearButton component", () => {
  test("renders", () => {
    const component = renderer.create(<ClearButton />);

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
