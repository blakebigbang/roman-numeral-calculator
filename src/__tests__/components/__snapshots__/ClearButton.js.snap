// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearButton component renders 1`] = `
<button
  className="css-1p3tv29"
  onClick={[Function]}
  tabIndex={0}
>
  AC
</button>
`;
